src/embind/embind.js

```diff
@@ -57,6 +57,9 @@ var LibraryEmbind = {
     // TODO: do we need a deleteObject here?  write a test where
     // emval is passed into JS via an interface
   }`,
+  $EmValOptionalType__deps: ['$EmValType'],
+  $EmValOptionalType: undefined,
+  $EmValOptionalType__postset: 'EmValOptionalType = Object.assign({optional: true}, EmValType);',
```

Comment:
I think there is a better way to do this.  Does ` $EmValOptionalType: '=Object.assign({optional: true}, EmValType);` work?

Reply:
That works! Much nicer.